File: backend/mmr/src/main/java/cit/edu/mmr/dto/ProfileDTO.java
package cit.edu.mmr.dto;

import jakarta.persistence.Lob;

public class ProfileDTO {
    private long userId;
    private String username;
    private String email;
    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBiography() {
        return biography;
    }

    public void setBiography(String biography) {
        this.biography = biography;
    }

    private String biography;

    @Lob
    private byte[] profilePicture;
    private java.util.Date createdAt;
    private String role;
    private boolean isOauthUser;

    // Getters and setters
    public long getUserId() { return userId; }
    public void setUserId(long userId) { this.userId = userId; }

    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public byte[] getProfilePicture() {
        return profilePicture;
    }

    public void setProfilePicture(byte[] profilePicture) {
        this.profilePicture = profilePicture;
    }

    public java.util.Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(java.util.Date createdAt) { this.createdAt = createdAt; }

    public String getRole() { return role; }
    public void setRole(String role) { this.role = role; }

    public boolean isOauthUser() { return isOauthUser; }
    public void setOauthUser(boolean oauthUser) { isOauthUser = oauthUser; }
}

File: frontend-mobile/Memoire/app/src/main/java/com/example/memoire/models/ProfileDTO.kt
package com.example.memoire.models

import java.util.Date


data class ProfileDTO(
    val id: Long? = null,
    val username: String = "",
    val name: String? = null,
    val email: String? = null,
    val biography: String? = null,
    val profilePicture: String? = null,
    val isActive: Boolean = true
)


data class ProfileDTO2(
    val userId: Long? = null,
    val username: String = "",
    val name: String? = null,
    val email: String? = null,
    val biography: String? = null,
    val profilePicture: String? = null,
    val isActive: Boolean = true
)